### PR TITLE
Refresh/new homepage

### PR DESCRIPTION
**Pull request by breezyfasano**
## Description & motivation
- Adding the refreshed homepage branded for Devhub
  - New components: `Card`, `BlogPostCard`, `Callout`, `PostCarousel`, and `Hero`

**[Preview link here](https://deploy-preview-2163--docs-getdbt-com.netlify.app/)**

## To-do before merge
- [x] Merge in Winnie's community page changes and resolve merge conflicts
- [x] Hook up the blog data and create the featured resource section
- [x] @emilymermell-dbt to QA & design review!
- [ ] Confirm with @runleonarun and @jasnonaz where the "Getting started guide" card URL will point to

### CODE REVIEW

**Comment by emilymermell-dbt** [comment]
Hi! This is looking great. [Here's my feedback video ](https://www.loom.com/share/2da3a370f4e041ce9875f44d39c08f2d)and I've attached the screenshot that I mentioned in my video in Asana. 

Overall notes:

- The image looks blurry, is there anything we can do to get it clearer?
- Can we remove the navy final CTA row at the bottom of this page?

See my CSS changes below:

.card_zHBt {
border-radius: 10px;
padding: 2.5rem 2.5rem 1.5rem 2.5rem;
}

.Hero__R2D {
min-height: 375px;
margin-bottom: 6rem;
}

.home .resource-section {
row-gap: 25px;
}

.row .col.col--8 {
--ifm-col-width: 60%;
}

**Comment by emilymermell-dbt** [comment]
Hi! This is looking great. [Here's my feedback video ](https://www.loom.com/share/2da3a370f4e041ce9875f44d39c08f2d)and I've attached the screenshot that I mentioned in my video in Asana. 

Overall notes:

- The image looks blurry, is there anything we can do to get it clearer?
- Can we remove the navy final CTA row at the bottom of this page?

See my CSS changes below:

.card_zHBt {
border-radius: 10px;
padding: 2.5rem 2.5rem 1.5rem 2.5rem;
}

.Hero__R2D {
min-height: 375px;
margin-bottom: 6rem;
}

.home .resource-section {
row-gap: 25px;
}

.row .col.col--8 {
--ifm-col-width: 60%;
}

**Review by john-rock — CHANGES_REQUESTED** [review]
This is looking awesome Breezy, nice work. I have two small things I noticed during review.

1. https://www.loom.com/share/6de845ccefb9406bab3d9090b858431f - See video for explanation of card icon behavior. Figure it would be easier to do a recording than explaining here. Let me know if you have any questions.

2. It looks like the postCarousel could use some bottom padding on the swiper. The individual posts are having there shadow and bottom border cut off on both light/dark modes
dbt_Docs
dbt_Docs

File: website/src/pages/index.js
Comment (JKarlavige) [inline website/src/pages/index.js:31]: Nice 😎

File: website/src/pages/index.js
Comment (JKarlavige) [inline website/src/pages/index.js]: This card is going to a 404 page, I believe the link here would be `https://www.getdbt.com/community/join-the-community/`

**Review by JKarlavige — CHANGES_REQUESTED** [review]
Nice work, looking great! Echoing John's comment about the slider cards being cut off on the bottom. Also, left one note for the "Join the community" card which 404s. All else is looking good!

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
            <h2>From the dbt Community</h2>
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  link="/community/join"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  title="Become a contributor"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  link="/community/contribute"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  title="Open source dbt Packages"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]: Hmm docs team just moved the getting started tutorial out of guides

Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  link="/guides/best-practices"
```

File: website/src/pages/index.js
Comment (JKarlavige) [reply website/src/pages/index.js]: Oh, just noticed there's a new `community/join` page in a separate Docs PR: https://deploy-preview-2069--docs-getdbt-com.netlify.app/community/join. So it might point to this instead?

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                  link="/community/forum"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]: todo: I'll work with @runleonarun to determine where this link should be pointed with the move of getting started guide from Guides to docs

**Review by jasnonaz — CHANGES_REQUESTED** [review]
This is looking incredible thank you Breezy! 😎

Couple notes around wording / links changing around. Appreciate you bearing with us as we finalize the exact locations of various content 🙃

File: website/src/pages/index.js
Comment (breezyfasano) [reply website/src/pages/index.js]: Yep that is it! :D

Comment (JKarlavige) [reply website/src/pages/index.js]: Awesome, then @jasnonaz's suggestion looks like it'll be correct for that card!

File: website/src/pages/index.js
Comment (breezyfasano) [reply website/src/pages/index.js]: Sounds good, I'll add this as a todo item before merge!

**Comment by breezyfasano** [comment]
Wooo, thank you SO much everyone! All feedback has been addressed + I've merged in Winnie's changes from `next` 🥳 

@john-rock, thank you for catching that icon bug! It's so strange, I can replicate on the preview link but not on `localhost`. I'm going to merge without fixing that though. The hook used to determine what mode a user is on is `useColorMode`, and it is from Docusaurus itself. So I would likely have to rebuild that hook which would be a lot to do for tomorrow. I've talked to Emily and we're cool with launching without fixing that, but I'll log an Asana task for it. 👍 

Just missing some the getting started card link from Ganz and Leona, but I'll get it from them via Slack.
Thanks everybody!

**Comment by john-rock** [comment]
> I'm going to merge without fixing that though. The hook used to determine what mode a user is on is `useColorMode`, and it is from Docusaurus itself. So I would likely have to rebuild that hook which would be a lot to do for tomorrow. I've talked to Emily and we're cool with launching without fixing that, but I'll log an Asana task for it. 👍 

Sounds good and totally agree not worth holding up merging because of it!

**Review by john-rock — APPROVED** [review]
lgtm!

File: website/src/pages/index.js
Comment (JKarlavige) [inline website/src/pages/index.js]: This link 404s when clicking on it. It looks like there's a redirect setup for this which goes to `/docs/supported-data-platforms`. Since it's a SPA, clicking the card doesn't hit the server, causing it to 404. But if you go directly to it, the redirect will fire, redirecting it to `/docs/supported-data-platforms`. Same issue that we ran into in the past for this site.

So updating this link to `/docs/supported-data-platforms` should fix it!

**Review by JKarlavige — CHANGES_REQUESTED** [review]
Couple small things. I see the TODO is already setup for the guides/getting-started link so all good there.

Added one note about the "Supported data platforms" card 404ing. Also, it looks like the bottom padding added to the slider is causing the arrows to be pushed down with it. Can be move the arrows back up to be in the middle of the cards?

image

All else is looking good!

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                    link="/docs/get-started/getting-started/overview"
```

File: website/src/pages/index.js
Comment (jasnonaz) [inline website/src/pages/index.js]:
```suggestion
                    link="/docs/build/projects"
```

**Review by jasnonaz — CHANGES_REQUESTED** [review]
This is looking great! Last couple link tweaks. THANK YOU BREEZY

File: website/src/pages/index.js
Comment (breezyfasano) [reply website/src/pages/index.js]: Thank you so much!

**Review by JKarlavige — APPROVED** [review]
lgtm, awesome work! :shipit:

**Review by jasnonaz — APPROVED** [review]
Incredible work, thank you Breezy!!

**Pull request merged**
